lib/icalendar/util/deserialize.ex

```diff
@@ -12,29 +12,61 @@ defmodule ICalendar.Util.Deserialize do
   end
 
   @doc ~S"""
-  iex> ICalendar.Util.Deserialize.retrieve_kvs("lorem:ipsum")
-  {"LOREM", "ipsum"}
+  This function extracts the key and value parts from each line of a iCalendar
+  string.
+
+      iex> ICalendar.Util.Deserialize.retrieve_kvs("lorem:ipsum")
+      {"LOREM", %{params: %{}, value: "ipsum"}}
   """
   def retrieve_kvs(line) do
+    # Split Line up into key and value
     [key, value] = String.split(line, ":", parts: 2, trim: true)
-    {String.upcase(key), value}
+    [key, params] = retrieve_params(key)
+
+    {String.upcase(key), %{value: value, params: params}}
```

Comment:
Is it worth using a struct here? Labels could be label, value, and params? I rather like structs.